openapi-cli/src/main/resources/templates/utils.bal

```diff
@@ -0,0 +1,239 @@
+import ballerina/url;
+
+type Encoding record {
+    # Defines how multiple values are delimited
+    string style = FORM;
+    # Specifies whether arrays and objects should generate as separate fields
+    boolean explode = true;
+};
+
+enum EncodingStyle {
+    DEEPOBJECT,
+    FORM,
+    SPACEDELIMITED,
+    PIPEDELIMITED
+}
+
+type SimpleBasicType string|boolean|int|float|decimal;
+
+final Encoding & readonly defaultEncoding = {};
+
+# Generate client request when the media type is given as application/x-www-form-urlencoded.
+#
+# + encodingMap - Includes the information about the encoding mechanism
+# + anyRecord - Record to be serialized
+# + return - Serialized request body or query parameter as a string
+isolated function createFormURLEncodedRequestBody(record {|anydata...; |} anyRecord, map<Encoding> encodingMap = {}) returns string {
+    string[] payload = [];
+    foreach [string, anydata] [key, value] in anyRecord.entries() {
+        Encoding encodingData = encodingMap.hasKey(key) ? encodingMap.get(key) : defaultEncoding;
+        if value is SimpleBasicType {
+            payload.push(key, "=", getEncodedUri(value.toString()));
+        } else if value is SimpleBasicType[] {
+            payload.push(getSerializedArray(key, value, encodingData.style, encodingData.explode));
+        } else if (value is record {}) {
+            if encodingData.style == DEEPOBJECT {
+                payload.push(getDeepObjectStyleRequest(key, value));
+            } else {
+                payload.push(getFormStyleRequest(key, value));
+            }
+        } else if (value is record {}[]) {
+            payload.push(getSerializedRecordArray(key, value, encodingData.style, encodingData.explode));
```

Comment:
@indikasampath2000 added new method named `getSerializedRecordArray` to handle `record{}[]` logic. Please review.

Reply:
Looks good.